src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -807,14 +826,16 @@ private static GapicProductConfig createDummyInstance(
     // pattern field has been removed. Therefore, throw an error if a single resource config is
     // found that does not exist in an annotation.
     for (String key : singleResourceNameConfigsFromGapicConfig.keySet()) {
-      if (!annotationResourceNameConfigs.containsKey(key)) {
+      String annotationsStyleKey = Name.from(key).toUpperCamel();
```

Comment:
Can you add a comment here about why converting to upper case is needed?

Reply:
Added. In brief: this is because `singleResourceNameConfigsFromGapicConfig` gets resource names from gapic_yaml, where they are in `lower_snake_case`, and `annotationResourceNameConfigs` has resources from annotations, where they are in `UpperCamelCase` so we need to convert the keys to common format before comparing them.